qiita_db/sql_connection.py

```diff
@@ -402,9 +430,16 @@ def execute_fetchall(self, sql, sql_args=None):
         """
         with self._sql_executor(sql, sql_args) as pgcursor:
             result = pgcursor.fetchall()
-        return result
+            if return_types:
+                types = {desc[0]: self.TYPE_CODES[desc[1]]
+                         for desc in pgcursor.description}
+
+        if return_types:
+            return result, types
```

Comment:
How as a seperate method though? You need to have the cursor to get the info, andthat only exists inside that `with`